src/main/java/uk/gov/hmcts/reform/sendletter/tasks/UploadLettersTask.java

```diff
@@ -52,17 +56,24 @@ public void run() {
 
         try (Stream<Letter> stream = repo.findByStatus(LetterStatus.Created)) {
             Iterator<Letter> iterator = stream.iterator();
+            int counter = 0;
+
             while (iterator.hasNext()) {
                 Letter letter = iterator.next();
 
                 try {
                     uploadLetter(letter);
+                    counter++;
                 } catch (FtpException exception) {
                     logger.error(String.format("Exception uploading letter %s", letter.getId()), exception);
                     break;
                 }
             }
 
+            if (counter > 0) {
```

Comment:
Could we not also report zero letters uploaded so we can see the task is running?

Reply:
I don't mean as a separate event, just removing the if.

Reply:
isn't log event sufficient enough to assume task is running? there are several logs happening during:
```java
// scheduler
String name = task.name();
log.info("Trying to lock {}", name);
log.info("Acquired lock {}", name);
// task
logger.info("Started letter upload job");
logger.info("Completed letter upload job");
// back to scheduler
log.info("Released lock {}", name);
```

Reply:
I'm wondering how it looks in app insights but can't seem to see any events in there to guide me.

My understanding is that logging is not available in app insights, so if you saw that this event was not occurring you would not know if it was because there is no work to do or because there is a problem with the task. Accessing logs is a pain, so it would be nice if App insights gave us a good overview of system health.

It may be nothing needs to be done here since i don't know what app insights looks like yet so I defer to you.